simplify.go

```diff
@@ -0,0 +1,72 @@
+package irutil
+
+import (
+	"math/big"
+
+	"github.com/llir/llvm/ir/constant"
+)
+
+// Simplify returns an equivalent (and potentially simplified) constant to
+// the constant expression.
+func Simplify(c constant.Constant) constant.Constant {
+	switch c := c.(type) {
+	case *constant.ExprAdd:
+		x, ok := Simplify(c.X).(*constant.Int)
+		y, ok2 := Simplify(c.Y).(*constant.Int)
+		if ok && ok2 {
+			x.X = (&big.Int{}).Add(x.X, y.X)
+		}
+		return x
```

Comment:
Perhaps this would work:

```go
	x, ok := Simplify(c.X).(*constant.Int)
	y, ok2 := Simplify(c.Y).(*constant.Int)
	if ok && ok2 {
		z := constant.NewInt(c.Typ, 0)
		z.X = z.X.Add(x.X, y.X)
		return z
	}
	return c
```